rest/docs/design/technical_design.md

```diff
@@ -0,0 +1,127 @@
+
+##### Connection
+It can be seen as a specific instance of `Connector`.  
+On the `Connector` entity you were just declaring the fields that the specific `Connector` supported, here you are passing the corresponding values for those fields.
+For example it's here that you specify that the value for the field `user` is `scott` and the value for the `password` field is `tiger`.
+
+##### Action
+`Action` represents a specific operation within the scope of a `Connector`.  
+In the Twitter example, an operation might be `postMessage()`.  
+An `Action` has its own set of of properties.  
+In the Twitter example, a property for an action might be `salute`, that a user might fill in with a value like `hello` or `good bye`.    
+Role of an `Action` is also to define the link with physical java code library that will provide connectivity with a specific system/technology:  
+`CamelConnectorGAV`. In this field, you specify the coordinates of the `Camel Connector` component that this `Action` delegates its logical operations to.
+
+
+##### Step
+`Step` is a **generic entity**, that represents a pipeline operation in your `Integration`. It might be a transformation, a logging call or anything else.
+Due to all this flexibility `Step` is actually a semantically poor entity. Everything but its type is optional: it can have references to an `Action` or a `Connection` but it doesn't have to. That fact that it needs them or not depends on its `type`, that is its most important information.
+An important type of `Step` is `Endpoint`. Unluckily, an enum that lists these kinds does not exist; So the relationship is weak. You have to take care to use a type that the framework is able to understand, because there is no preemptive way to validate your choice.
+Steps are declared within an `Integration`.
+
+
+The overall idea is that your `Integration` define 2 kind of information:
+1. a catalog of technologies and allowed operations (`Connectors`,  `Actions`)
+2. a pipeline of customized components(`Connection`) and operations you need to perform(`Step`), using the building blocks defined at 1).
+
+To reinforce the idea you can see it as if:
+
+You define some (technology)`Connector`, where you declare all the properties you allow the developer to give a value to in a corresponding set of `Connection`s. You also link this `Connector` with a series of operations that are tied to this technology, represented by `Action`. In each `Action` you declare also eventual additional properties.  
+At last, you detail how the data will flow across `Connections`, and with which eventual tranformation, specifying a list of `Steps`.
+
+Notice that despite you define an ordered list of `Steps` to define the business behavior of your `Integration` there is no type checking of any sort, between those steps. It's up to you be sure that the output of a `Step` is coherent with the output of the following one.
```

Comment:
is there way to check for this input/output? Would be nice to add how to that here, what do you think?